.github/workflows/checks.yml

```diff
@@ -185,17 +232,250 @@ jobs:
+          cmake \
+          -D CMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE \
+          -D CMAKE_INSTALL_PREFIX='/usr/lib/freecad' \
+          -D CMAKE_INSTALL_BINDIR='/usr/lib/freecad/bin' \
+          -D CMAKE_INSTALL_LIBDIR='/usr/lib/freecad/lib' \
+          -D CMAKE_INSTALL_DATADIR='/usr/share/freecad' \
+          -D CMAKE_INSTALL_DATAROOTDIR='/usr/share' \
+          -D CMAKE_INSTALL_DOCDIR='/usr/share/doc/freecad' \
+          -B build/ >> ${{ env.logdir }}Install.log 2>> ${{ env.logdir }}InstallErrors.log
+
+          # Build again to copy the files to the install path
+          cmake --build build/ >> ${{ env.logdir }}Install.log 2>> ${{ env.logdir }}InstallErrors.log
+
+          # Run CMake Install
+          sudo cmake --install build/ >> ${{ env.logdir }}Install.log 2>> ${{ env.logdir }}InstallErrors.log
+
+          exitCode=$?
+
+          # links for bin
+          FILES=/usr/lib/freecad/bin/*
+          for f in $FILES
+          do
+            sudo ln -vs /usr/lib/freecad/bin/$(basename $f) /usr/bin/$(basename $f)
+          done
+
+          # Write the install report
+          if [ $exitCode -eq 0 ] ; then
+            echo "<details><summary>:heavy_check_mark: CMake install succeeded</summary>" >> ${{ env.logdir }}checkReport.md
+          else
+            echo "<details><summary>:fire: CMake install failed</summary>" >> ${{ env.logdir }}checkReport.md
+          fi
+          echo "" >> ${{ env.logdir }}checkReport.md
+          echo "Install Error Log (stderr output):" >> ${{ env.logdir }}checkReport.md
+          echo '```' >> ${{ env.logdir }}checkReport.md
+          cat ${{ env.logdir }}InstallErrors.log >> ${{ env.logdir }}checkReport.md
+          echo '```' >> ${{ env.logdir }}checkReport.md
+          echo "Install Error Log (stdout output trimmed to the last 100 Lines):" >> ${{ env.logdir }}checkReport.md
+          echo '```' >> ${{ env.logdir }}checkReport.md
+          tail -n 100 ${{ env.logdir }}Install.log >> ${{ env.logdir }}checkReport.md
+          echo '```' >> ${{ env.logdir }}checkReport.md
+          echo "</details>">> ${{ env.logdir }}checkReport.md
+          echo "" >> ${{ env.logdir }}checkReport.md
+
+          # Print the Log to the console
+          cat ${{ env.logdir }}InstallErrors.log
+          echo "::group::Install Log"
+          cat ${{ env.logdir }}Install.log
+          echo "::endgroup::"
+
+          # Exit the step with the exit code of the Install
+          if [ ${{ env.cmakeInstallFailSilent }} != true ]; then
+            exit $exitCode
+          else
+            echo "CMake install exited with code $exitCode, but we are ignoring it"
+            exit 0
+          fi
+
+      - name: FreeCAD UnitTests on Install
+        id: testInstall
+        if: env.testInstall == 'true' && always()
+        run: |
+          set +e
+
+          LD_LIBRARY_PATH=/usr/lib/freecad/lib FreeCADCmd -t 0 &> ${{ env.logdir }}TestInstall.log
```

Comment:
consider running the GUI tests instead, see @0penBrain's PR #6421 for how to do this, also why are we running the checks twice?

Reply:
I tried to run the GUI checks, but with the Ubuntu 22.04 runner Xvfb gives me problems 😞 . On Ubuntu 20.04, Xvfb works fine. It's on my to-do list, but I wanted to get this out quickly after my last botched PR. (And as of next week, I'm on vacation) 
Maybe the first testing can be omitted, I left it in because cmake install + testing on the install was giving me problems and I can easily turn it on and off that way. And the 5min extra runtime is negligible after we don't have to pay for GitHub actions anyway.

Reply:
I'm not sure if we want to run on Ubuntu 22.04 it has a broken version of pivy and I doubt they'll be updating that. About the double testing, IMO it would clutter output so it's not convenient to us to begin with and besides just because a service is free doesn't mean you have to abuse it, let's be a little considerate with GitHub and not waste CPU time for nothing.